cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiff.java

```diff
@@ -139,7 +139,7 @@ private static boolean isIgnorableProperty(String key) {
     private static Collection<AlterConfigOp> diff(int brokerId, String desired,
                                                   Config brokerConfigs,
                                                   Map<String, ConfigModel> configModel) {
-        if (brokerConfigs == null) {
+        if (brokerConfigs == null || desired == null) {
```

Comment:
It might be worth documenting a null desired state does not mean an empty broker state but is indeed a shortcut for saying no changes required.
This can not be assumed and should be in the javadoc